examples/forecaster/Transformer Forecaster demo.ipynb

```diff
@@ -0,0 +1,640 @@
+    "    FUTURE context is empty. This is in accordance with how CRAFT Model was\n",
+    "    originally trained on CGA-CMV, taking the last context from each\n",
+    "    conversation (\"last\" defined as being up to and including the chronologically\n",
+    "    last utterance as recorded in the corpus)\n",
+    "    \"\"\"\n",
+    "    matches_split = (context_tuple.current_utterance.get_conversation().meta[\"split\"] == \"train\")\n",
+    "    is_end = (len(context_tuple.future_context) == 0)\n",
+    "    return (matches_split and is_end)\n",
+    "def eval_selector(context_tuple):\n",
+    "\n",
+    "    return (context_tuple.current_utterance.get_conversation().meta[\"split\"] == \"val\")\n",
+    "def transform_selector(context_tuple):\n",
+    "    \"\"\"\n",
+    "    For transform we only need to check that the conversation is in the test split\n",
+    "    \"\"\"\n",
+    "    return (context_tuple.current_utterance.get_conversation().meta[\"split\"] == \"test\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9614aff5-843e-4b3b-b03f-6f57f8e76b8a",
+   "metadata": {},
+   "source": [
+    "## Initialize the Forecaster and TransformerEncoderModel backend\n",
+    "\n",
+    "Now the rest of the process is pretty straightforward! We simply need to:\n",
+    "1. Initialize a backend `ForecasterModel` for the `Forecaster` to use, in this case we choose ConvoKit's implementation of TransformerEncoderModel.\n",
+    "2. Initialize a `Forecaster` instance to wrap that `ForecasterModel` in a generic fit-transform API"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "9ee62d7f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# CPU mode (noting that it will be slower)\n",
+    "DEVICE = \"cuda\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "653febf7-ef59-4e8b-8ce2-39b4a33c3747",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Some weights of RobertaForSequenceClassification were not initialized from the model checkpoint at roberta-base and are newly initialized: ['classifier.dense.bias', 'classifier.dense.weight', 'classifier.out_proj.bias', 'classifier.out_proj.weight']\n",
+      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
+     ]
+    }
+   ],
+   "source": [
+    "model_name_or_path = 'roberta-base'\n",
+    "\"\"\"\n",
+    "List of models that we use this class to benchmark:\n",
+    "bert-base-cased\n",
+    "bert-large-cased\n",
+    "roberta-base\n",
+    "roberta-large\n",
+    "SpanBERT/spanbert-base-cased\n",
+    "SpanBERT/spanbert-large-cased\n",
+    "microsoft/deberta-v3-base\n",
+    "microsoft/deberta-v3-large\n",
+    "\"\"\"\n",
+    "\n",
+    "config_dict = ForecasterTrainingArgument(\n",
+    "                    output_dir=\"YOUR_SAVING_DIRECTORY\",\n",
```

Comment:
The literal string "YOUR_SAVING_DIRECTORY" is repeated many times throughout the notebook, and presumably the intent is that the end user replace _every single instance_ of it with a single actual directory name. But someone going through the notebook for this first time might not realize there are multiple places that need to be changed. The proper way to handle it is to just make this a variable, and have the user just update that variable once (as opposed to having to find-and-replace this string literal).